src/lib/utils/enhanceConfiguration.ts

```diff
@@ -13,7 +15,10 @@ function enhanceConfiguration(
   // Get the relevant partial configuration asked by the widget
   const partialConfiguration = widget.getConfiguration(configuration);
 
-  return mergeDeep(configuration, partialConfiguration);
+  return mergeSearchParameters(
+    new algoliasearchHelper.SearchParameters(configuration),
```

Comment:
We could have updated the signature to avoid the creation of this `SearchParameters` on each call.

```ts
type enhance = (configuration: SearchParameters, widget: Widget) => SearchParameters
```

Reply:
And call the constructor before passing it to enhance?

Reply:
Yes.